src/controllers/controller.polarArea.js

```diff
@@ -138,6 +138,32 @@ module.exports = function(Chart) {
 			});
 		},
 
+		getStartAngle: function(customAngles, dataSetData, metaData, datasetStartAngle, circumference, index) {
+
+			// If there is NaN data before us, we need to calculate the starting angle correctly.
+			// We could be way more efficient here, but its unlikely that the polar area chart will have a lot of data
+
+			var previousAngle = 0;
+			var visibleCount = 0;
+			var startAngle = 0;
+			var i = 0;
+
+			for (i = 0; i < index; ++i) {
+				if (customAngles !== undefined) {
+					previousAngle += customAngles[visibleCount];
+				}
+				if (!isNaN(dataSetData[i]) && !metaData[i].hidden) {
+					++visibleCount;
+				}
+			}
+			if (customAngles === undefined) {
```

Comment:
Could this condition be placed before the loop and return immediately start angle to avoid looping if not necessary? Moreover, this would avoid the customAngles check in the loop.